### PR TITLE
Add DocuSign Click connector

### PR DESCRIPTION
**Pull request by sachinira**
# Description
Generate the DocuSign Click connector using the official OpenAPI specification. 
https://developers.docusign.com/docs/click-api/reference/

Fixes: https://github.com/ballerina-platform/ballerina-extended-library/issues/248

One line release note: 
- Generate DocuSign Click connector using Open API specification.

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Not tested yet

**Test Configuration**:
* Ballerina Version: SL Beta 6
* Operating System: Ubuntu 20.04
* Java SDK: 11

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

### Changes to connector module

- [x] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [ ] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Comment by Chuhaa** [comment]
@sachinira CI build fails due to old versions of dependencies. Could you regenerate dependencies using https://github.com/ballerina-platform/ballerina-distribution/suites/4918117688/artifacts/141807451?

File: openapi/docusign.click/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["Content & Files/Documents", "Cost/Freemium"]
+org = "ballerinax"
+name = "docusign.click"
+distribution = "slbeta6"
+repository = "https://github.com/ballerina-platform/ballerinax-openapi-connectors"
+version = "0.1.0"
```
Comment (RolandHewage) [inline openapi/docusign.click/Ballerina.toml:8]: Shall we update the version to 1.0.0

File: openapi/docusign.click/client.bal [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    # + return - An error if connector initialization failed 
+    public isolated function init(ClientConfig clientConfig, string serviceUrl = "https://www.docusign.net/clickapi") returns error? {
+        http:Client httpEp = check new (serviceUrl, clientConfig);
+        self.clientEp = httpEp;
+        return;
+    }
+    # Gets the current version and other information about the Click API.
+    #
+    # + return - Successful response. 
+    remote isolated function serviceInformationGetServiceinformation() returns ServiceInformation|error {
```
Comment (RolandHewage) [inline openapi/docusign.click/client.bal:72]: If possible please change the method names in a meaningful way.

File: openapi/docusign.click/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,1645 @@
+openapi: 3.0.1
+info:
+  title: DocuSign Click API
+  description: DocuSign Click lets you capture consent to standard agreement terms with a single click terms and conditions, terms of service, terms of use, privacy policies, and more. The Click API lets you include this customizable clickwrap solution in your DocuSign integrations.
```
Comment (RolandHewage) [inline openapi/docusign.click/openapi.yaml:4]: Shall we follow the standard used in other connectors in the description as follows
https://github.com/ballerina-platform/openapi-connectors/blob/49164bb7baae2ddcd19037c109548143c0fe32d7/openapi/ably/openapi.yaml#L14
https://github.com/ballerina-platform/openapi-connectors/blob/49164bb7baae2ddcd19037c109548143c0fe32d7/openapi/spotify/openapi.yaml#L7

**Review by RolandHewage — COMMENTED** [review]
Requested few changes.

**Review by RolandHewage — APPROVED** [review]

File: openapi/docusign.click/Package.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+Connects to [DocuSign Click API](https://metamind.readme.io/reference#predictive-vision-service-api) from Ballerina.
```
Comment (aneeshafedo) [inline openapi/docusign.click/Package.md:1]: URL is not given is not directing to `DocuSign Click API`. Please verify

**Review by abeykoon — APPROVED** [review]

**Pull request merged**